### PR TITLE
More Snippets!

### PR DESCRIPTION
**Pull request by cmoesel**
This adds snippets for Logical, Resource, Invariant, Mapping, and RuleSet (see README for recommended trigger phrases).

This also updates the tab-stops to allow the user to change auto-inserted text placeholders (whereas before it skipped them).

### CODE REVIEW

File: snippets/fsh-snippets.json [hunk 1]
```diff
@@ -16,21 +16,47 @@
...
 		],
 		"description": "Create a FSH Extension"
 	},
 
+	"FSH Logical": {
+		"prefix": ["log", "Log"],
+		"body": [
+			"Logical: $1",
+			"Parent: ${2:Base}",
+			"Id: ${3:$1}",
```
Comment (jafeltra) [inline snippets/fsh-snippets.json:32]: Is the `Id` for Logicals typically the same as the name, or is it usually kebab cased (I think that's what it's called), similar to how the Profile and Extension snippet works?

File: snippets/fsh-snippets.json [hunk 2]
```diff
@@ -16,21 +16,47 @@
...
+		],
+		"description": "Create a FSH Resource"
+	},
+
+	"FSH Resource": {
+		"prefix": ["res", "Res"],
+		"body": [
+			"Resource: $1",
+			"Parent: ${2|DomainResource,Resource|}",
+			"Id: ${3:$1}",
```
Comment (jafeltra) [inline snippets/fsh-snippets.json:45]: Similar to the question above, is this usually the same as the Resource name or does it also get kebab cased like the Profile and Extension ids?

File: snippets/fsh-snippets.json [hunk 3]
```diff
@@ -40,24 +66,61 @@
...
+	},
+
+	"FSH Mapping": {
+		"prefix": ["map", "Map"],
+		"body": [
+			"Mapping: $1",
+			"Source: $2",
+			"Target: \"$3\"",
+			"Id: \"$4\"",
+			"Title: \"${5:${4/(([A-Za-z]*)[^A-Za-z0-9]([A-Za-z]*))|(([A-Za-z]*[a-z0-9]+)([A-Za-z]+))|(([A-Za-z]+)([A-Za-z0-9])([a-z]))/${2:/capitalize}$5$8 ${3:/capitalize}$6$9$10/g}}\"",
```
Comment (jafeltra) [inline snippets/fsh-snippets.json:111]: image

It seems like maybe the auto-update for `Title` on Mappings is off a little. When I fill in `Id` and hit tab, the last character gets separated to a new word.

**Review by jafeltra — COMMENTED** [review]
This looks good! I had a couple small questions, but I'm very excited about all these snippets to use when I forget which keywords go where.

File: snippets/fsh-snippets.json (hunk 1 above)
Comment (cmoesel) [reply snippets/fsh-snippets.json:32]: Yeah, for logicals and resources it seems the id is usually the same as the name (for whatever reason).  It's a little less clear for logicals but more clear for resources.  For example, the id for `MedicationRequest` is `MedicationRequest`, not `medication-request`.  In many cases, logicals and resources are similar, so I'm sort of leaning into that here.

File: snippets/fsh-snippets.json (hunk 2 above)
Comment (cmoesel) [reply snippets/fsh-snippets.json:45]: As noted in my other comment, I think it's usually the same.

File: snippets/fsh-snippets.json (hunk 3 above)
Comment (cmoesel) [reply snippets/fsh-snippets.json:111]: Dang.  Mapping is where I tried to get fancy and gave up.  Maybe I left in some of the broken "fancy" code by mistake.  Thanks!

**Comment by cmoesel** [comment]
@jafeltra - I tried to get it to do what I wanted, but... I just couldn't.  So now the mapping title does nothing fancy.  But at least it is not broken!

**Review by jafeltra — APPROVED** [review]
Works for me! Sometimes we just have to let `Description` fields grow up and find their own path.

**Pull request merged**